packages/main/src/RatingIndicator.ts

```diff
@@ -174,10 +78,106 @@ const metadata = {
  * @public
  * @since 1.0.0-rc.8
  */
+
+@customElement("ui5-rating-indicator")
+@languageAware
+
+/**
+ * The event is fired when the value changes.
+ *
+ * @event sap.ui.webc.main.RatingIndicator#change
+ * @public
+ */
+@event("change")
+
 class RatingIndicator extends UI5Element {
-	static get metadata() {
-		return metadata;
-	}
+	/**
+	 * The indicated value of the rating.
+	 * <br><br>
+	 * <b>Note:</b> If you set a number which is not round, it would be shown as follows:
+	 * <ul>
+	 * <li>1.0 - 1.2 -> 1</li>
+	 * <li>1.3 - 1.7 -> 1.5</li>
+	 * <li>1.8 - 1.9 -> 2</li>
+	 * <ul>
+	 * @type {sap.ui.webc.base.types.Float}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.value
+	 * @defaultvalue 0
+	 * @public
+	 */
+	@property({ validator: Float, defaultValue: 0 })
+	value!: number;
+
+	/**
+	 * The number of displayed rating symbols.
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.max
+	 * @defaultvalue 5
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property({ validator: Integer, defaultValue: 5 })
+	max!: number;
+
+	/**
+	 * Defines whether the component is disabled.
+	 *
+	 * <br><br>
+	 * <b>Note:</b> A disabled component is completely noninteractive.
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.disabled
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	disabled!: boolean;
+
+	/**
+	 * Defines whether the component is read-only.
+	 * <br><br>
+	 * <b>Note:</b> A read-only component is not editable,
+	 * but still provides visual feedback upon user interaction.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.readonly
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	readonly!: boolean;
+
+	/**
+	 * Defines the accessible ARIA name of the component.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.accessibleName
+	 * @defaultvalue: undefined
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property()
+	accessibleName!: string;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	_stars!: Array<Star>;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	_focused!: boolean;
+
+	/**
+	 * Indicates when to fire change event
+	 * @private
+	 */
+	@property({ validator: Float, defaultValue: null })
+	_liveValue?: number;
```

Comment:
previously it was not used as metadata property. defining it in that way may do unwanted invalidations. for me best solution is to make code working with number and undefined only and removing the decorator.

Reply:
Done.